### PR TITLE
progress made, almost done

### PR DESCRIPTION
**Pull request by jghil**

### CODE REVIEW

File: express-notes-api/index.js [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+app.get('/api/notes', (req, res) => {
+  const newNotes = [];
+  for (const property in dataJSON.notes) {
+    newNotes.push(dataJSON.notes[property]);
+  }
+  if (newNotes.length !== 0) {
+    res.status(200).json(newNotes);
+  } else {
+    res.json([newNotes]);
+  }
```
Comment (scbowler) [inline express-notes-api/index.js:16]: If there are no notes then `newNotes` will be an empty array, so there is no need for an `if` statement here. You can just do:
```JAVASCRIPT
res.json(newNotes);
```

Also if you are sending a status of `200` you don't have to explicitly set that, `200` is the default status code.

File: express-notes-api/index.js [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+  } else if (dataJSON.notes[id]) {
+    res.status(200).json(dataJSON.notes[id]);
+  } else {
+    res.status(404).json({ error: 'cannot find notes with id: ' + id });
+  }
+});
+
+app.post('/api/notes', (req, res) => {
+  const body = req.body;
+  if (body === undefined) {
```
Comment (scbowler) [inline express-notes-api/index.js:32]: Here you want to check for the `content` property specifically:
```JAVASCRIPT
if (body.content === undefined) {
```

File: express-notes-api/index.js [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+    });
+  }
+});
+
+app.put('/api/notes/:id', (req, res) => {
+  const id = Number(req.params.id);
+  const body = req.body;
+  if (!Number.isInteger(id) || id <= 0) {
+    res.status(400).json({ error: 'id must be a positive integer!' });
+  } else if (body === undefined) {
```
Comment (scbowler) [inline express-notes-api/index.js:79]: Same here, you want to check for the `content` property:
```JAVASCRIPT
} else if (body.content === undefined) {
```

**Review by scbowler — CHANGES_REQUESTED** [review]
Looking good, just a few updates needed. Let me know if you have any questions.

File: express-notes-api/index.js (hunk 1 above)
Comment (jghil) [reply express-notes-api/index.js:16]: so I could take the whole `if` statement and `else` statement out and replace the code with `res.json(newNotes);`?

**Pull request merged**